linter/lints/rfc/lint_crl_has_valid_timestamps.go

```diff
@@ -0,0 +1,244 @@
+ */
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+	"golang.org/x/crypto/cryptobyte"
+	cryptobyte_asn1 "golang.org/x/crypto/cryptobyte/asn1"
+)
+
+const (
+	utcTimeFormat         = "YYMMDDHHMMSSZ"
+	generalizedTimeFormat = "YYYYMMDDHHMMSSZ"
+)
+
+type crlHasValidTimestamps struct{}
+
+/************************************************
+RFC 5280: 5.1.2.4
+CRL issuers conforming to this profile MUST encode thisUpdate as UTCTime for
+dates through the year 2049. CRL issuers conforming to this profile MUST encode
+thisUpdate as GeneralizedTime for dates in the year 2050 or later. Conforming
+applications MUST be able to process dates that are encoded in either UTCTime or
+GeneralizedTime.
+
+Where encoded as UTCTime, thisUpdate MUST be specified and interpreted as
+defined in Section 4.1.2.5.1. Where encoded as GeneralizedTime, thisUpdate MUST
+be specified and interpreted as defined in Section 4.1.2.5.2.
+
+RFC 5280: 5.1.2.5
+CRL issuers conforming to this profile MUST encode nextUpdate as UTCTime for
+dates through the year 2049. CRL issuers conforming to this profile MUST encode
+nextUpdate as GeneralizedTime for dates in the year 2050 or later. Conforming
+applications MUST be able to process dates that are encoded in either UTCTime or
+GeneralizedTime.
+
+Where encoded as UTCTime, nextUpdate MUST be specified and interpreted as
+defined in Section 4.1.2.5.1. Where encoded as GeneralizedTime, nextUpdate MUST
+be specified and interpreted as defined in Section 4.1.2.5.2.
+
+RFC 5280: 5.1.2.6
+The time for revocationDate MUST be expressed as described in Section 5.1.2.4.
+
+RFC 5280: 4.1.2.5.1
+UTCTime values MUST be expressed in Greenwich Mean Time (Zulu) and MUST include
+seconds (i.e., times are YYMMDDHHMMSSZ), even where the number of seconds is
+zero.
+
+RFC 5280: 4.1.2.5.2
+GeneralizedTime values MUST be expressed in Greenwich Mean Time (Zulu) and MUST
+include seconds (i.e., times are YYYYMMDDHHMMSSZ), even where the number of
+seconds is zero. GeneralizedTime values MUST NOT include fractional seconds.
+************************************************/
+
+func init() {
+	lint.RegisterRevocationListLint(&lint.RevocationListLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_crl_has_valid_timestamps",
+			Description:   "CRL thisUpdate, nextUpdate, and revocationDates must be properly encoded",
+			Citation:      "RFC 5280: 5.1.2.4, 5.1.2.5, and 5.1.2.6",
+			Source:        lint.RFC5280,
+			EffectiveDate: util.RFC5280Date,
+		},
+		Lint: NewCrlHasValidTimestamps,
+	})
+}
+
+func NewCrlHasValidTimestamps() lint.RevocationListLintInterface {
+	return &crlHasValidTimestamps{}
+}
+
+func (l *crlHasValidTimestamps) CheckApplies(c *x509.RevocationList) bool {
+	return true
+}
+
+func (l *crlHasValidTimestamps) Execute(c *x509.RevocationList) *lint.LintResult {
+	input := cryptobyte.String(c.RawTBSRevocationList)
```

Comment:
It was really hard to review this diff to verify that all the substantive lines were moved and not changed. Note for other reviewers: here's how I did it:

```git show 55a8dc75b3ca872d161e80f97d12d32c5bd4d132:linter/lints/rfc/lint_crl_has_valid_timestamps.go > /tmp/new
git show 55a8dc75b3ca872d161e80f97d12d32c5bd4d132~:linter/lints/crl/lints.go > /tmp/old
vimdiff /tmp/{old,new}
```

I also manually edited `old` while diffing, to delete a lot of the head of lints.go, which helped the differ figure things out.